File: Background.cpp
#include "Background.h"



Background::Background(sf::Vector2u myWinSize)
{
	licznik = 0;
	if (!forestTexture.loadFromFile("forest.jpg"))
		std::cout << "Blad las!" << std::endl;
	if (!skyTexture.loadFromFile("sky.jpg"))
		std::cout << "Blad niebo!" << std::endl;
	if (!grassTexture.loadFromFile("grass.jpg"))
		std::cout << "Blad trawa!" << std::endl;

	for (int i = 0; i < 13; i++,licznik++)
	{
		tabSprite.push_back(mySprite);
		tabSprite[licznik].setTexture(forestTexture);
		tabSprite[licznik].setPosition(-6 * float(myWinSize.x) + i * float(myWinSize.x), 0);
	}
	for (int i = 0; i < 13; i++, licznik++)
	{
		tabSprite.push_back(mySprite);
		tabSprite[licznik].setTexture(skyTexture);
		tabSprite[licznik].setPosition(-6 * float(myWinSize.x) + i * float(myWinSize.x), -float(myWinSize.y));
	}
	for (int i = 0; i < 13; i++, licznik++)
	{
		tabSprite.push_back(mySprite);
		tabSprite[licznik].setTexture(skyTexture);
		tabSprite[licznik].setPosition(-6 * float(myWinSize.x) + i * float(myWinSize.x), -2*float(myWinSize.y));
	}
	for (int i = 0; i < 13; i++, licznik++)
	{
		tabSprite.push_back(mySprite);
		tabSprite[licznik].setTexture(grassTexture);
		tabSprite[licznik].setPosition(-6 * float(myWinSize.x) + i * float(myWinSize.x), float(myWinSize.y));
	}
}

/*void Background::draw(sf::RenderTarget& target, sf::RenderStates states) const
{
	for (int i = 0; i < 15; i++) {
		target.draw(tabSprite[i]);
	}
}*/

void Background::displayGraphics(sf::RenderWindow& myWindow, std::vector<Player> players, Arrow liveArrow, std::vector<DeadArrow> deadarrows)
{
	for (int i = 0; i < 52; i++) {
		myWindow.draw(tabSprite[i]);
	}
	for (int i = 0; i<players.size(); i++)
	{
		myWindow.draw(players[i]);
	}
	if (!liveArrow.isDead) {
		myWindow.draw(liveArrow);
	}
	for (int i = 0; i < deadarrows.size(); i++) {
		myWindow.draw(deadarrows[i]);
	}
}

void Background::displayBackground(sf::RenderWindow& myWindow)
{
	for (int i = 0; i < 52; i++)
	{
		myWindow.draw(tabSprite[i]);
	}
}

File: Background.java
package pl.ArcherJ.Main;

import java.io.IOException;
import java.nio.file.Paths;
import java.util.Vector;
import org.jsfml.graphics.RenderWindow;
import org.jsfml.graphics.Sprite;
import org.jsfml.graphics.Texture;


public class Background {
    int spriteTabIterator;
    Texture forestTexture;
    Texture skyTexture;
    Texture grassTexture;
    Sprite mySprite;
    Vector <Sprite> tabSprite;
    
    Background(RenderWindow myWindow) throws IOException{
        spriteTabIterator = 0;
        tabSprite=new Vector<Sprite>();

        forestTexture=new Texture();
        forestTexture.loadFromFile(Paths.get("forest.jpg"));
        skyTexture=new Texture();
        skyTexture.loadFromFile(Paths.get("sky.jpg"));
        grassTexture=new Texture();
        grassTexture.loadFromFile(Paths.get("grass.jpg"));

        for (int i = 0; i < 13; i++, spriteTabIterator++)
        {
            tabSprite.add(mySprite);
            tabSprite.set(spriteTabIterator, new Sprite(forestTexture));
            tabSprite.get(spriteTabIterator).setPosition(-6 * (float)(myWindow.getSize().x) + i * (float)(myWindow.getSize().x), 0);
        }
        for (int i = 0; i < 13; i++, spriteTabIterator++)
        {
            tabSprite.add(mySprite);
            tabSprite.set(spriteTabIterator, new Sprite(skyTexture));
            tabSprite.get(spriteTabIterator).setPosition(-6 * (float)(myWindow.getSize().x) + i * (float)(myWindow.getSize().x), -(float)(myWindow.getSize().y));
        }
        for (int i = 0; i < 13; i++, spriteTabIterator++)
        {
            tabSprite.add(mySprite);
            tabSprite.set(spriteTabIterator, new Sprite(skyTexture));
            tabSprite.get(spriteTabIterator).setPosition(-6 * (float)(myWindow.getSize().x) + i * (float)(myWindow.getSize().x), -2 * (float)(myWindow.getSize().y));
        }
        for (int i = 0; i < 13; i++, spriteTabIterator++)
        {
            tabSprite.add(mySprite);
            tabSprite.set(spriteTabIterator, new Sprite(grassTexture));
            tabSprite.get(spriteTabIterator).setPosition(-6 * (float)(myWindow.getSize().x) + i * (float)(myWindow.getSize().x), (float)(myWindow.getSize().y));
        }
    }
    public void displayBackground(RenderWindow myWindow){
        for (int i = 0; i < 52; i++)
        {
            myWindow.draw(tabSprite.get(i));
        }
    }
}
